lib/public/resource.rb

```diff
@@ -203,8 +203,13 @@ def dump
       }
     end
 
-    sig { params(spec: T::Hash[String, T.untyped]).returns(Resource) }
-    def self.load(spec)
+    sig do
+      params(
+        spec: T::Hash[String, T.untyped],
+        configuration: ResourceRegistry::Configuration
+      ).returns(Resource)
+    end
+    def self.load(spec, configuration: ResourceRegistry.configuration)
```

Comment:
In the future I see space to have a Factory class only for this purpose, but let's keep evolving things :)